modules/core/02-client/legacy/v100/store.go

```diff
@@ -0,0 +1,69 @@
+package v100
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/cosmos/ibc-go/modules/core/02-client/types"
+	host "github.com/cosmos/ibc-go/modules/core/24-host"
+	"github.com/cosmos/ibc-go/modules/core/exported"
+	ibctmtypes "github.com/cosmos/ibc-go/modules/light-clients/07-tendermint/types"
+)
+
+// MigrateStore performs in-place store migrations from SDK v0.40 of the IBC module to v1.0.0 of ibc-go.
+// The migration includes:
+//
+// - Pruning solo machine clients
```

Comment:
Since the proto definition changed, the first thought of @AdityaSripal and I is to delete the solo machine, but I'm realizing there could be connection in INIT attached to that solo machine. I could delete those connections as well

Alternatively, I could add back solomachine proto definition v1. Unmarshal into that, copy over the fields to the new proto definition and set the updated proto in state. Requires adding back v1 of the solo machine proto definition

Thoughts? @AdityaSripal @cwgoes @fedekunze

Reply:
Are there any opened channels using solo machines? If so, it seems like deleting clients would be a problem.

I don't think channels in INIT state are an issue, those can be deleted as well, but if any are open that's a problem.

Reply:
No channels. The connection handshake failed in OpenTry/OpenAck previously

Reply:
OK, subject to that holding true this seems safe to me, although isn't it always possible that channels could be created in between when this code is deployed and when the switchover actually happens (e.g. if the upgrade module is used)?

Reply:
No I don't think so. Unless a chain upgraded to an unreleased commit of this code and/or backported the fixes we added. The previous version being upgraded from could never create an open channel with solo machines. When this upgrade occurs, the migrations will be run first (same idea as init genesis but for x/upgrade)

Reply:
Ah yes - sorry, I had forgotten that this wasn't possible at the moment anyways - makes sense & seems safe to me.

Reply:
updated, my only question now is about channel capabilities. The logic for removing connections/channels turned out good

- remove connection if client does not exist
- remove channel if connection does not exist

Reply:
see updated changes. Decided migrating from v1 to v2 was the better option